src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs

```diff
@@ -137,13 +142,759 @@ string DomainMapper(Match match)
+        public void Docs_EngineCapabilities_LazyQuantifiers(RegexOptions options)
+        {
+            const string Input = "This sentence ends with the number 107325.";
+
+            const string GreedyPattern = @".+(\d+)\.";
+            Match match = Regex.Match(Input, GreedyPattern, options);
+            Assert.True(match.Success);
+            Assert.Equal("5", match.Groups[1].Value);
+
+            const string LazyPattern = @".+?(\d+)\.";
+            match = Regex.Match(Input, LazyPattern, options);
+            Assert.True(match.Success);
+            Assert.Equal("107325", match.Groups[1].Value);
+        }
+
+        // https://docs.microsoft.com/en-us/dotnet/standard/base-types/details-of-regular-expression-behavior#net-framework-engine-capabilities
+        [Theory]
+        [InlineData(RegexOptions.None)]
+        [InlineData(RegexOptions.Compiled)]
+        public void Docs_EngineCapabilities_PositiveLookahead(RegexOptions options)
+        {
+            const string Pattern = @"\b[A-Z]+\b(?=\P{P})";
+            const string Input = "If so, what comes next?";
+            MatchCollection matches = Regex.Matches(Input, Pattern, RegexOptions.IgnoreCase | options);
+            Assert.Equal(3, matches.Count);
+            Assert.Equal("If", matches[0].Value);
+            Assert.Equal("what", matches[1].Value);
+            Assert.Equal("comes", matches[2].Value);
+        }
+
+        // https://docs.microsoft.com/en-us/dotnet/standard/base-types/details-of-regular-expression-behavior#net-framework-engine-capabilities
+        [Theory]
+        [InlineData(RegexOptions.None)]
+        [InlineData(RegexOptions.Compiled)]
+        public void Docs_EngineCapabilities_NegativeLookahead(RegexOptions options)
+        {
+            const string Pattern = @"\b(?!non)\w+\b";
+            const string Input = "Nonsense is not always non-functional.";
+            MatchCollection matches = Regex.Matches(Input, Pattern, RegexOptions.IgnoreCase | options);
+            Assert.Equal(4, matches.Count);
+            Assert.Equal("is", matches[0].Value);
+            Assert.Equal("not", matches[1].Value);
+            Assert.Equal("always", matches[2].Value);
+            Assert.Equal("functional", matches[3].Value);
+        }
+
+        // https://docs.microsoft.com/en-us/dotnet/standard/base-types/details-of-regular-expression-behavior#net-framework-engine-capabilities
+        [Theory]
+        [InlineData(RegexOptions.None)]
+        [InlineData(RegexOptions.Compiled)]
+        public void Docs_EngineCapabilities_ConditionalEvaluation(RegexOptions options)
+        {
+            const string Pattern = @"^(?<Pvt>\<PRIVATE\>\s)?(?(Pvt)((\w+\p{P}?\s)+)|((\w+\p{P}?\s)+))\r?$";
+
+            string input = "<PRIVATE> This is not for public consumption." + Environment.NewLine +
+                           "But this is for public consumption." + Environment.NewLine +
+                           "<PRIVATE> Again, this is confidential.\n";
```

Comment:
Can we avoid keywords like confidential? Credscan is running in the official builds and fails when it fines suspicious phrases that might leak a token.

Reply:
Ok